crates/router/src/core/payment_link.rs

```diff
@@ -56,32 +70,47 @@ pub async fn intiate_payment_link_flow(
         .get_required_value("fulfillment_time")
         .change_context(errors::ApiErrorResponse::PaymentNotFound)?;
 
-    helpers::validate_payment_status_against_not_allowed_statuses(
-        &payment_intent.status,
-        &[
-            storage_enums::IntentStatus::Cancelled,
-            storage_enums::IntentStatus::Succeeded,
-            storage_enums::IntentStatus::Processing,
-            storage_enums::IntentStatus::RequiresCapture,
-            storage_enums::IntentStatus::RequiresMerchantAction,
-        ],
-        "create payment link",
-    )?;
+    let payment_link_metadata = merchant_account
+        .payment_link_metadata
+        .map(|pl_metadata| {
+            serde_json::from_value::<admin_types::PaymentLinkMetadata>(pl_metadata)
+                .into_report()
+                .change_context(errors::ApiErrorResponse::InvalidDataValue {
+                    field_name: "payment_link_metadata",
+                })
+        })
+        .transpose()?;
 
-    let expiry = fulfillment_time.assume_utc().unix_timestamp();
+    let order_details = payment_intent
+        .order_details
+        .get_required_value("order_details")
+        .change_context(errors::ApiErrorResponse::MissingRequiredField {
+            field_name: "order_details",
+        })?;
 
-    let js_script = get_js_script(
-        payment_intent.amount.to_string(),
-        payment_intent.currency.unwrap_or_default().to_string(),
-        merchant_account.publishable_key.unwrap_or_default(),
-        payment_intent.client_secret.unwrap_or_default(),
-        payment_intent.payment_id,
-        expiry,
-    );
+    let payment_details = api_models::payments::PaymentLinkDetails {
+        amount: payment_intent.amount,
+        currency: payment_intent.currency.unwrap_or_default(),
+        payment_id: payment_intent.payment_id,
+        merchant_name: merchant_account.merchant_name,
+        order_details,
+        return_url: payment_intent.return_url.unwrap_or_default(),
+        expiry: fulfillment_time,
+        pub_key: merchant_account.publishable_key.unwrap_or_default(),
+        client_secret: payment_intent.client_secret.unwrap_or_default(),
+        merchant_logo: payment_link_metadata
+            .clone()
+            .map(|pl_metadata| pl_metadata.merchant_logo.unwrap_or_default())
+            .unwrap_or_default(),
```

Comment:
why so many unwrap_or_default? can you handle the cases where these will be None? or if there is no such case then these can be made as mandatory, another case to handle this is to raise the error if value is missing